packages/flutter/test/widgets/slivers_padding_test.dart

```diff
@@ -423,4 +424,32 @@ void main() {
       const Rect.fromLTRB(0.0, -200.0, 800.0, 200.0),
     );
   });
+
+  testWidgets('SliverPadding includes preceding padding in the precedingScrollExtent provided to child', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/49195
+    final GlobalKey key = GlobalKey();
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: CustomScrollView(
+        slivers: <Widget>[
+          SliverPadding(
+            padding: const EdgeInsets.only(top: 30),
+            sliver: SliverFillRemaining(
+              hasScrollBody: false,
+              child: Container(
+                key: key,
+                color: Colors.red,
+              ),
+            )
+          ),
+        ]
+      ),
+    ));
+    await tester.pump();
+
+    expect(
+      tester.renderObject<RenderBox>(find.byKey(key)).size.height,
+      equals(570),
```

Comment:
Can you add a comment why this has to be 570? (as per style guide avoid magic numbers)

Reply:
Sure thing! :)

Reply:
Let me know if this makes better sense now.

Reply:
Wonderful. I would have also expected something shorter, like:

```
// height of viewport - top padding = 600 - 30 = 570.
```

:)

Reply:
Yeah. It's a bit --verbose. 😆